tests/support/whl_from_dir/whl_from_dir_repo.bzl

```diff
@@ -0,0 +1,50 @@
+"""Creates a whl file from a directory tree.
+
+Used to test wheels. Avoids checking in prebuilt files and their associated
+security risks.
+"""
+
+load("//python/private:repo_utils.bzl", "repo_utils")  # buildifier: disable=bzl-visibility
+
+def _whl_from_dir_repo(rctx):
+    root = rctx.path(rctx.attr.root).dirname
+    repo_utils.watch_tree(rctx, root)
+
+    output = rctx.path(rctx.attr.output)
+    repo_utils.execute_checked(
+        rctx,
+        # cd to root so zip recursively takes everything there.
+        working_directory = str(root),
+        op = "WhlFromDir",
+        arguments = [
+            "zip",
+            "-0",  # Skip compressing
+            "-X",  # Don't store file time or metadata
+            str(output),
+            "-r",
+            ".",
+        ],
+    )
+    rctx.file("BUILD.bazel", 'exports_files(glob(["*"]))')
+
+whl_from_dir_repo = repository_rule(
```

Comment:
I remember somebody also requested a feature similar to this and wanted to add `PROJECT_ROOT`. Maybe this thing would be useful to them?

Ref: #2517

Reply:
Yeah, I had a similar thought. I spent some time yesterday thinking about different ways to map custom sources for dependencies into the pip integration. I think https://github.com/bazel-contrib/rules_python/issues/2986 is about the closest we have for this topic, so I'll post details there.